datadog_checks_dev/datadog_checks/dev/tooling/templates/docs/default.yaml

```diff
@@ -0,0 +1,199 @@
+# TODO: To be split into separate files
+
+- template: overview
+  name: Overview
+  header_level: 2
+  description: |
+    TODO: General comments
+  parameters:
+  prepend_text:
+  append_text:
+  processor:
+  hidden: false
+  sections:
+
+- template: setup
+  name: Setup
+  header_level: 2
+  description: |
+    Follow the instructions below to install and configure this check for an Agent running on a host. For
+    containerized environments, see the [Autodiscovery Integration Templates][1] for guidance on applying these
+    instructions.
+
+    [1]: https://docs.datadoghq.com/agent/kubernetes/integrations/
```

Comment:
Haven't though about how, but links should definitely be a construct with its own logic

Reply:
The thought was for each section to be self-contained with its link definitions.  So it could define either via embedded or reference style as shown here.  During the Producer phase, they would all be changed to embedded, then changed back to reference style as part of the README consumer.

This makes it easy to create and edit sections and templates, since they are just snippets of Markdown.

An extension to this could be an extra `links-ref` parameter (naming TBD) as described in Additional Information above, but it shouldn't be required.